lib/IRGen/GenClass.cpp

```diff
@@ -471,7 +393,14 @@ namespace {
       // of a global field offset variable which will be initialized by
       // either the Objective-C or Swift runtime, depending on the
       // class's heritage.
-      return FieldAccess::NonConstantDirect;
+      if (Options.contains(ClassMetadataFlags::ClassHasMissingMembers) ||
+          Options.contains(ClassMetadataFlags::ClassHasResilientMembers) ||
+          Options.contains(ClassMetadataFlags::ClassHasObjCAncestry))
```

Comment:
…or if the class has resilient ancestry.

Reply:
Just like in the previous case I still think this is right; "resilient ancestry" means we have ancestors whose metadata size is not known at compile time (they can add or remove vtable entries!) but we might still know the stored property layout (they might all be @_fixed_layout). Sorry :)